src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
> 1. there would be practically no chance of hitting this.

Hitting what? 

a) Expecting culture-sensitive result on en-US, but getting ordinal result?
b) Expecting the same speedup as en-US on a different culture when data is in ASCII range?